### PR TITLE
[wip] Adjust code to changed kube_derive

### PR DESCRIPTION
**Pull request by kesselborn**
This tries to fix deislabs/krustlet#573

The kube-derive macro now implements `kube::Resource` instead of
`k8s_openapi::Resource` and and `k8s_openapi::Metadata`.

I think these adjustments should be ok, `however: kube::Resource` does not
have the method `metadata_mut()` and I couldn't fix the `resync` method
@kflansburg

I added the compilation of the example app to the github actions
workflow so this kind of error get get caught early

### CODE REVIEW

**Comment by kflansburg** [comment]
Thanks for tackling this. Didn't realize it would be this complex. 

* Good call adding it to the CI tests.
* Looks like the `metadata_mut` issue will be addressed in upcoming `kube` release https://github.com/clux/kube-rs/pull/486. 
* Good observation that we don't really need `k8s_openapi::Resource` trait bound anymore. 

I'll take a crack at eliminating the `metadata_mut` entirely.

**Review by kflansburg — APPROVED** [review]
@kesselborn I have removed the need for `metadata_mut()` by simplifying the channel used in the runtime module. 

If this looks good to you, I think we can merge.

File: crates/krator/src/runtime.rs [hunk 1]
```diff
@@ -90,16 +113,16 @@ impl<O: Operator> OperatorRuntime<O> {
...
-        match &event {
-            Event::Applied(object) => {
-                let key: ObjectKey = object.into();
+    async fn dispatch(&mut self, event: ObjectEvent<O::Manifest>) -> anyhow::Result<()> {
+        match event {
+            ObjectEvent::Applied(object) => {
+                let key: ObjectKey = (&object).into();
                 // We are explicitly not using the entry api here to insert to avoid the need for a
                 // mutex
                 match self.handlers.get_mut(&key) {
```
Comment (kesselborn) [inline crates/krator/src/runtime.rs:122]: a `get` should be enough here, shouldn't it?

File: crates/krator/src/runtime.rs [hunk 2]
```diff
@@ -55,13 +54,37 @@ impl<R: Resource> From<&Event<R>> for PrettyEvent {
...
+                namespace: object.namespace(),
+            },
+            ObjectEvent::Deleted { name, namespace } => PrettyEvent::Deleted {
+                name: name.to_string(),
+                namespace: namespace.clone(),
+            },
+        }
+    }
+}
+
```
Comment (kesselborn) [inline crates/krator/src/runtime.rs:80]: I know it wasn't introduced here, but what is `PrettyEvent` actually used for? Is this just for tracing purposes?

**Review by kesselborn — COMMENTED** [review]
lgtm

File: crates/krator/src/runtime.rs (hunk 2 above)
Comment (kflansburg) [reply crates/krator/src/runtime.rs:80]: Yes, just makes it easier to debug events without dumping the entire object spec.

File: crates/krator/src/runtime.rs (hunk 1 above)
Comment (kflansburg) [reply crates/krator/src/runtime.rs:122]: I think you are right, Tokio 0.3 made it so you don't need a mutable reference when calling `Sender::send`. Surprised Clippy doesnt complain.

**Pull request merged**